### PR TITLE
Disable non-const reference check from cpplint

### PR DESCRIPTION
**Pull request by jcwchen**
### Description

Disable non-const reference check from cpplint.

### Motivation and Context

Preferred coding-style: over pointers, when non-null.

### CODE REVIEW

**Review by jcwchen — COMMENTED** [review]
> Could you say more on why we want to do it for my learning? Thanks!

This check seems debatable and there is more context in cpplint repo: https://github.com/cpplint/cpplint/issues/148. Usually I did see quite a few warnings from this check when reviewing ONNX PRs. @gramalingam suggested we can disable this check.

**Comment by justinchuby** [comment]
Thank you!

**Comment by gramalingam** [comment]
To add to what Jacky said: we prefer to "Use a non-const reference for arguments that are modifiable but cannot be nullptr so the API clearly advertises the intent" ... this is a variation from the Google style that onnxruntime also adopts (see: https://github.com/microsoft/onnxruntime/blob/main/docs/Coding_Conventions_and_Standards.md )

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -48,6 +48,7 @@ jobs:
           reporter: github-pr-check
           level: warning
           flags: --linelength=120
+          filter: "-runtime/references"
```
Comment (gramalingam) [inline .github/workflows/lint.yml:51]: Is this for this specific error-message? Just wondering, since the name sounds too generic. Thanks!

**Review by gramalingam — APPROVED** [review]

**Review by xadupre — APPROVED** [review]

File: .github/workflows/lint.yml (hunk 1 above)
Comment (justinchuby) [reply .github/workflows/lint.yml:51]: I think so: https://github.com/cpplint/cpplint/issues/106

**Comment by andife** [comment]
> To add to what Jacky said: we prefer to "Use a non-const reference for arguments that are modifiable but cannot be nullptr so the API clearly advertises the intent" ... this is a variation from the Google style that onnxruntime also adopts (see: https://github.com/microsoft/onnxruntime/blob/main/docs/Coding_Conventions_and_Standards.md )

I wonder if we shouldn't officially document our code style as well? And create a similar document for Onnx itself?

**Pull request merged**